coalahtml/_coalahtml/app/controllers/files.js

```diff
@@ -33,7 +33,28 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
-      return path_name.split('/').slice(-1)[0];
+
+    var getValidPath = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/');
+    };
+    $scope.basename = function(path_name) {
+      return getValidPath(path_name).slice(-1)[0];
+    };
+    var _slice = function(path_name, start, stop) {
+      return getValidPath(path_name).slice(start, stop);
+    };
+    var len = function(path_name) {
+      return getValidPath(path_name).length;
+    };
```

Comment:
You have a whole bunch of functions that basically call one method. Also, you split path_name every time so it'll have slightly worse performance.